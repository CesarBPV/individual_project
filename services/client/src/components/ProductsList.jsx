import React from 'react';


const ProductsList = (props) => {
  return (
    <div>
      {
        props.products.map((product) => {
          return (
            <h4
              key={product.id}
              className="box title is-4"
            >{ product.name }
            </h4>
          )
        })
      }
    </div>
  )
};


export default ProductsList;