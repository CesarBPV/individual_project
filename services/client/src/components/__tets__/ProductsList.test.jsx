import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
 
import ProductsList from '../ProductsList';
 
const products = [
  {
    'active': true,
    'stock': 20,
    'price': 0.7,
    'trademark': 'Nabisco',
    'category': 'Galletas',
    'id': 1,
    'name': 'Oreo'
  },
  {
    'active': true,
    'stock': 25,
    'price': 0.8,
    'trademark': 'Costa',
    'category': 'Galletas',
    'id': 2,
    'name': 'Picaras Fresa'
  }
];
 
test('ProductsList renders properly', () => {
  const wrapper = shallow(<ProductsList products={products}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('Oreo');
});

test('ProductsList renders a snapshot properly', () => {
  const tree = renderer.create(<ProductsList products={products}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
