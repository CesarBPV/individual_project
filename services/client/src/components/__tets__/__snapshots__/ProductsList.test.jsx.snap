// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    Oreo
  </h4>
  <h4
    className="box title is-4"
  >
    Picaras Fresa
  </h4>
</div>
`;
